connexion/validators/form_data.py

```diff
@@ -1,85 +1,59 @@
 import logging
 import typing as t
 
-from jsonschema import Draft4Validator, ValidationError, draft4_format_checker
-from starlette.datastructures import FormData, Headers, UploadFile
+from jsonschema import ValidationError, draft4_format_checker
+from starlette.datastructures import Headers, UploadFile
 from starlette.formparsers import FormParser, MultiPartParser
-from starlette.types import Receive, Scope
+from starlette.types import ASGIApp, Scope
 
 from connexion.exceptions import BadRequestProblem, ExtraParameterProblem
-from connexion.json_schema import Draft4RequestValidator
+from connexion.json_schema import Draft4RequestValidator, format_error_with_path
 from connexion.uri_parsing import AbstractURIParser
-from connexion.utils import is_null
+from connexion.validators import AbstractRequestBodyValidator
 
 logger = logging.getLogger("connexion.validators.form_data")
 
 
-class FormDataValidator:
+class FormDataValidator(AbstractRequestBodyValidator):
     """Request body validator for form content types."""
 
     def __init__(
         self,
-        scope: Scope,
-        receive: Receive,
+        next_app: ASGIApp,
         *,
         schema: dict,
-        validator: t.Type[Draft4Validator] = None,
         required=False,
         nullable=False,
         encoding: str,
-        uri_parser: t.Optional[AbstractURIParser] = None,
         strict_validation: bool,
+        uri_parser: t.Optional[AbstractURIParser] = None,
     ) -> None:
-        self._scope = scope
-        self._receive = receive
-        self.schema = schema
-        self.has_default = schema.get("default", False)
-        self.nullable = nullable
-        self.required = required
-        validator_cls = validator or Draft4RequestValidator
-        self.validator = validator_cls(schema, format_checker=draft4_format_checker)
-        self.uri_parser = uri_parser
-        self.encoding = encoding
-        self._messages: t.List[t.MutableMapping[str, t.Any]] = []
-        self.headers = Headers(scope=scope)
-        self.strict_validation = strict_validation
-        self.check_empty()
+        super().__init__(
+            next_app,
+            schema=schema,
+            required=required,
+            nullable=nullable,
+            encoding=encoding,
+            strict_validation=strict_validation,
+        )
+        self._uri_parser = uri_parser
```

Comment:
Not completely happy with this. The `__init__` arguments for the validators are fixed in the `request_validation` , so it doesn't really make sense that individual validators define additional parameters.
On the other hand, we do support form data specifically, so I can live with it. Open for better solutions though.

Reply:
Don't have a better idea at the moment :)